### ISSUE TITLE
The number of training sets printed in the log does not match

### ISSUE THREAD

**Issue by srdfjy** [facebookresearch/svoice#44]
Hi，The actual number of my training set is 4000, the verification set and test set are 500, why the log print training set is 1157？

-----------------------------------------------------------------------------

[2021-09-24 09:56:01,729][svoice.solver][INFO] - Training...
[2021-09-24 09:59:01,476][svoice.solver][INFO] - Train | Epoch 60 | 231/1157 | 1.3 it/sec | Loss -8.01471
[2021-09-24 10:02:00,937][svoice.solver][INFO] - Train | Epoch 60 | 462/1157 | 1.3 it/sec | Loss -7.95500
[2021-09-24 10:05:00,119][svoice.solver][INFO] - Train | Epoch 60 | 693/1157 | 1.3 it/sec | Loss -7.93800
[2021-09-24 10:07:59,123][svoice.solver][INFO] - Train | Epoch 60 | 924/1157 | 1.3 it/sec | Loss -7.92135
[2021-09-24 10:11:00,928][svoice.solver][INFO] - Train | Epoch 60 | 1155/1157 | 1.3 it/sec | Loss -7.90971
[2021-09-24 10:11:02,425][svoice.solver][INFO] - Train Summary | End of Epoch 60 | Time 900.70s | Train Loss -7.90990
[2021-09-24 10:11:02,426][svoice.solver][INFO] - ----------------------------------------------------------------------
[2021-09-24 10:11:02,426][svoice.solver][INFO] - Cross validation...
[2021-09-24 10:11:13,715][svoice.solver][INFO] - Valid | Epoch 60 | 100/500 | 9.1 it/sec | Loss -11.36663
[2021-09-24 10:11:24,161][svoice.solver][INFO] - Valid | Epoch 60 | 200/500 | 9.3 it/sec | Loss -11.31748
[2021-09-24 10:11:35,182][svoice.solver][INFO] - Valid | Epoch 60 | 300/500 | 9.2 it/sec | Loss -11.50216
[2021-09-24 10:11:46,996][svoice.solver][INFO] - Valid | Epoch 60 | 400/500 | 9.0 it/sec | Loss -11.34704
[2021-09-24 10:11:58,362][svoice.solver][INFO] - Valid | Epoch 60 | 500/500 | 9.0 it/sec | Loss -11.42437
[2021-09-24 10:11:58,363][svoice.solver][INFO] - Valid Summary | End of Epoch 60 | Time 956.63s | Valid Loss -11.42437
[2021-09-24 10:11:58,364][svoice.solver][INFO] - Learning rate adjusted: 0.00027
[2021-09-24 10:11:58,364][svoice.solver][INFO] - ----------------------------------------------------------------------
[2021-09-24 10:11:58,365][svoice.solver][INFO] - Evaluating on the test set...
[2021-09-24 10:12:11,507][svoice.evaluate][INFO] - Eval estimates | 100/500 | 7.8 it/sec
[2021-09-24 10:12:24,636][svoice.evaluate][INFO] - Eval estimates | 200/500 | 7.7 it/sec
[2021-09-24 10:12:37,091][svoice.evaluate][INFO] - Eval estimates | 300/500 | 7.8 it/sec
[2021-09-24 10:12:46,929][svoice.evaluate][INFO] - Eval estimates | 400/500 | 8.3 it/sec
[2021-09-24 10:12:56,445][svoice.evaluate][INFO] - Eval estimates | 500/500 | 8.7 it/sec
[2021-09-24 10:12:56,446][svoice.evaluate][INFO] - Eval metrics | 100/500 | 210968.5 it/sec
[2021-09-24 10:12:56,447][svoice.evaluate][INFO] - Eval metrics | 200/500 | 148216.4 it/sec
[2021-09-24 10:12:56,448][svoice.evaluate][INFO] - Eval metrics | 300/500 | 151577.1 it/sec
[2021-09-24 10:12:56,448][svoice.evaluate][INFO] - Eval metrics | 400/500 | 157615.6 it/sec
[2021-09-24 10:12:56,449][svoice.evaluate][INFO] - Eval metrics | 500/500 | 160678.0 it/sec
[2021-09-24 10:12:56,494][svoice.evaluate][INFO] - Test set performance: SISNRi=11.10 PESQ=0.0, STOI=0.0.
[2021-09-24 10:12:56,525][svoice.solver][INFO] - Separate and save samples...
100%|█████████████████████████████████████████████████████████████████████████████████| 125/125 [00:28<00:00,  4.41it/s]
[2021-09-24 10:13:24,919][svoice.solver][INFO] - ----------------------------------------------------------------------
[2021-09-24 10:13:24,920][svoice.solver][INFO] - Overall Summary | Epoch 60 | Train -7.90990 | Valid -11.42437 | Best -11.55503 | Sisnr 11.10285 | Pesq 0.00000 | Stoi 0.00000

**Comment by adiyoss** [comment]
@srdfjy,
It depends on your batch-size, if you set it to the default value which is 4 than you should have 1000 mini-batchs. 
Moreover, in case you have files which are longer than the `segment` parameter value (default is 4), the dataloader will split it to several files using the `stride` parameter (default is 1). For instance, for a 6 sec utterance you will get 3 training examples 0-4, 1-5, 2-6. That will probably explain the additional 157 batches.

**Comment by srdfjy** [comment]
@adiyoss Thank you for your detailed reply.
